js/article.js

```diff
@@ -11,37 +11,37 @@ function Article (opts) {
   this.authorUrl = opts.authorUrl;
 }
 
-// USING JQUERY TO TO FILL IN TEMPLATE CLONE WITH PROPERTIES FROM THE PARTICULAR ARTICLE INSTANCE
+// ADD PROPERTIES USED BY TEMPLATE . EXCUTE LOGIC HERE SINCE TEMPLATE CAN'T HOLD JS LOGIC. ADD RESULT TO OBJECT AS NEW PROPERTY BY KEY IN TEMPLATE. //
 Article.prototype.toHtml = function() {
-  /* TODO: DONE- Now use jQuery t fill in the rest of the current template
-  clone with properties from this particular Article instance.
-  We need to fill in:
-  1. author name,
-  2. author url,
-  3. article title,
-  4. article body, and */
-
-  var $newArticle = $('article.template').clone();
-  console.log($newArticle);
-  // article category
-  $newArticle.attr('data-category', this.category);
-  // article author 
-  $newArticle.attr('data-author', this.author);
-  // article title
-  $newArticle.find('h1').text(this.title);
-  // article author
-  $newArticle.find('.byline a').text(this.author);
-  // article auth url
-  $newArticle.find('.byline a').attr('href', this.authorUrl);
-  // article body
-  $newArticle.find('.article-body').html(this.body);
-  // publication date
-  $newArticle.find('time[pubdate]').attr('title', this.publishedOn);
-  // display the date as a relative number of 'days ago'
-  $newArticle.find('time').html('about ' + parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000) + ' days ago');
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+  this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+  // TODO: DONE! Use handlebars to render your articles!
+  //       - Select your template from the DOM.
+  //       - Now "compile" your template with Handlebars.
+  //       - Don't forget to return your template for this article.
+  var source = $('#articles-template').html();
+  var templateRender = Handlebars.compile(source);
+  return templateRender(this);
+};
 
+  //  AUTHOR-FILTER FUNCTION //
```

Comment:
I bet that once you refactor this to be dryer, and pass through the toHtml function, you'll get your drop down menus working no problem.